### DISCUSSION TITLE
Could not get value from json string?

### DISCUSSION THREAD

**Discussion by maazullah96** [nlohmann/json#3054] [Q&A] [answered]
I am passing the data from command line as string but when i get it in the code the error occurs but when i use it as variable value it works fine .I could not understand why the error is occuring. just need a little help of reading the json data.
the data i am using is 
`[
   {
      "1632491516":[
         {
            "X":386,
            "Y":108
         },
         {
            "X":555,
            "Y":704
         },
         {
            "X":833,
            "Y":654
         },
         {
            "X":594,
            "Y":114
         },
         {
            "X":386,
            "Y":108
         }
      ]
   },
   {
      "1632491517":[
         {
            "X":386,
            "Y":108
         },
         {
            "X":555,
            "Y":704
         },
         {
            "X":833,
            "Y":654
         },
         {
            "X":594,
            "Y":114
         },
         {
            "X":386,
            "Y":108
         }
      ]
   }
]
ABOUT
The JSON For`
how I can iterate over this JSON data if passed from the command line

**Comment by nlohmann** [comment, marked as answer]
Did you parse the string? Then you can access the values just like in https://github.com/nlohmann/json#examples.

**Reply by maazullah96** [reply]
json toJson(const char* jsonString) {
	json jsonObj;
	std::stringstream(jsonString) >> jsonObj;
	return jsonObj;
}

![image](https://user-images.githubusercontent.com/30190807/135860283-3d77131a-e042-41ec-be47-25efee329831.png)
I am parsing string but get the error 

[json.exception.parse_error.101] parse error at line 1, column 38: syntax error while parsing array - invalid literal; last read: '"{"X'; expected ']'

![image](https://user-images.githubusercontent.com/30190807/135860502-bf888f32-1b46-4546-bd7b-e75398be3900.png)

**Reply by nlohmann** [reply]
- The parse error indicates that your input was not valid JSON. Looks like a quote too many.
- You can replace your `toJson` function with `json::parse`, see https://json.nlohmann.me/api/basic_json/parse/.

**Reply by maazullah96** [reply]
Thanks for your help I have fixed the issue .The error was because of invalid JSON .
